system/Commands/Translation/LocalizationFinder.php

```diff
@@ -0,0 +1,370 @@
+     */
+    private function writeIsVerbose(string $text = '', ?string $foreground = null, ?string $background = null): void
+    {
+        if ($this->verbose) {
+            CLI::write($text, $foreground, $background);
+        }
+    }
+
+    private function isSubDirectory(string $directory, string $rootDirectory): bool
+    {
+        return 0 === strncmp($directory, $rootDirectory, strlen($directory));
+    }
+
+    private function process(string $currentDir, string $currentLocale): void
+    {
+        $tableRows    = [];
+        $countNewKeys = 0;
+
+        $iterator = new RecursiveIteratorIterator(new RecursiveDirectoryIterator($currentDir));
+        $files    = iterator_to_array($iterator, true);
+        ksort($files);
+
+        [$languageFoundKeys, $countFiles] = $this->findLanguageKeysInFiles($files);
+        ksort($languageFoundKeys);
+
+        /**
+         * New translates
+         */
+        $languageDiff        = [];
+        $languageFoundGroups = array_unique(array_keys($languageFoundKeys));
+
+        foreach ($languageFoundGroups as $langFileName) {
+            $languageStoredKeys = [];
+            $languageFilePath   = $this->languagePath . DIRECTORY_SEPARATOR . $currentLocale . DIRECTORY_SEPARATOR . $langFileName . '.php';
+
+            if (is_file($languageFilePath)) {
+                /**
+                 * Load old localization
+                 */
+                $languageStoredKeys = require $languageFilePath;
+            }
+
+            $languageDiff = $this->arrayDiffRecursive($languageFoundKeys[$langFileName], $languageStoredKeys);
+            $countNewKeys += $this->arrayCountRecursive($languageDiff);
+
+            if ($this->showNew) {
+                $tableRows = array_merge($this->arrayToTableRows($langFileName, $languageDiff), $tableRows);
+            } else {
+                $newLanguageKeys = array_replace_recursive($languageFoundKeys[$langFileName], $languageStoredKeys);
+
+                /**
+                 * New translates exists
+                 */
+                if ($languageDiff !== []) {
+                    if (false === file_put_contents($languageFilePath, $this->templateFile($newLanguageKeys))) {
+                        $this->writeIsVerbose('Lang file ' . $langFileName . ' (error write).', 'red');
+                    } else {
+                        exec('composer cs-fix --quiet ' . $this->languagePath);
+                        $this->writeIsVerbose('Lang file "' . $langFileName . '" successful updated!', 'green');
+                    }
+                }
+            }
+        }
+
+        if ($this->showNew && ! empty($tableRows)) {
+            sort($tableRows);
+            CLI::table($tableRows, ['File', 'Key']);
+        }
+
+        $this->writeIsVerbose('Files found: ' . $countFiles);
+        $this->writeIsVerbose('New translates found: ' . $countNewKeys);
+    }
+
+    /**
+     * @param SplFileInfo[] $files
+     *
+     * @return array<int, array|int>
```

Comment:
```suggestion
     * @return array<int, array|int>
     * @phpstan-return list{0: array<string, array<string, string>>, 1: int}
```

Reply:
I tried to figure out the details of php stan, I couldn't